packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/pipeline-actions.tsx

```diff
@@ -84,7 +89,15 @@ export const PipelineActions: React.FunctionComponent<PipelineActionsProps> = ({
   );
 };
 
-export default connect(null, {
+const mapState = ({ pipeline }: RootState) => ({
+  isPipelineInvalid: pipeline
+    .filter((x) => x.isEnabled)
+    .some((x) => !x.isValid),
```

Comment:
You can do it all in one pass, especially as `filter` will go through the whole array and `some` have a change of immediately stopping when condition is met. A tiny thing here that might not need to be changed, but just want to encourage you to pay attention to things like that

```suggestion
  isPipelineInvalid: pipeline
    .some((x) => x.isEnabled && !x.isValid),
```